vxquery-core/src/main/java/org/apache/vxquery/compiler/rewriter/rules/ConvertToAlgebricksExpressionsRule.java

```diff
@@ -33,21 +31,24 @@
 import edu.uci.ics.hyracks.algebricks.core.algebra.base.IOptimizationContext;
 import edu.uci.ics.hyracks.algebricks.core.algebra.base.LogicalExpressionTag;
 import edu.uci.ics.hyracks.algebricks.core.algebra.expressions.AbstractFunctionCallExpression;
+import edu.uci.ics.hyracks.algebricks.core.algebra.expressions.ExpressionAnnotationNoCopyImpl;
+import edu.uci.ics.hyracks.algebricks.core.algebra.expressions.IExpressionAnnotation;
 import edu.uci.ics.hyracks.algebricks.core.algebra.functions.AlgebricksBuiltinFunctions;
 import edu.uci.ics.hyracks.algebricks.core.algebra.functions.FunctionIdentifier;
 import edu.uci.ics.hyracks.algebricks.core.algebra.functions.IFunctionInfo;
 import edu.uci.ics.hyracks.algebricks.core.rewriter.base.IAlgebraicRewriteRule;
 
 /**
- * The rule searches for where the XQuery function are used in place of Algebricks builtin function. 
- * The combination the boolean XQuery function and the XQuery equivalent function are replace with 
- * the Algebricks builtin function  .
+ * The rule searches for where the XQuery function are used in place of Algebricks builtin function.
+ * The combination of the boolean XQuery function and the XQuery equivalent function are replaced with
+ * the Algebricks builtin function .
  * 
  * <pre>
  * Before
  * 
  *   plan__parent
- *   %OPERATOR( $v1 : boolean(xquery_function( \@input_expression ) ) )
+ *   %OPERATOR( $v1 : boolean(xquery_function( \@input_expression ) or $v1 : general-compare(xquery_function( \@input_expression ) 
```

Comment:
How about this comment?
plan__parent
%OPERATOR( $v1 : boolean(xquery_function( \@input_expression ) ) )
plan__child

or 

plan__parent
%OPERATOR( $v1 : xquery_function( \@input_expression ) )
plan__child

where xquery_function has a known equivalent in Algebricks, such as conditional expressions and a check for null.